cmd/controller-manager/app/serve.go

```diff
@@ -32,11 +31,7 @@ import (
 	"k8s.io/kubernetes/pkg/util/configz"
 )
 
-type serveFunc func(handler http.Handler, shutdownTimeout time.Duration, stopCh <-chan struct{}) error
-
-// Serve creates a base handler chain for a controller manager. It runs the
-// the chain with the given serveFunc.
-func Serve(c *CompletedConfig, serveFunc serveFunc, stopCh <-chan struct{}) error {
+func DefaultBuildHandlerChain(c *CompletedConfig) http.Handler {
```

Comment:
take a handler in?  Trying to match `func(apiHandler http.Handler, c *Config) (secure http.Handler)`

Reply:
No, remove serverFunc. `DefaultBuildHandlerChain` only takes in config and return a handler like it does in apiserver.

Reply:
github does not compare accurately